documentation/testing-docs/test-matrix-for-az-accounts.md

```diff
@@ -0,0 +1,196 @@
+# Test Matrix for Authentication in Az.Accounts
+
+The authentication functionality in Az.Accounts is one of most important piece for Azure PowerShell, to make sure Azure PowerShell be delivered to customers with high quality, we define the test matrix and it should be honored by each release of Az.Accounts.
+
+**Priority Clarification**
+
+- `P0`: Run if any authentication related code change in Az.Accounts
+- `P1`: Run if upgrading minor version of Azure.Identity and/or MSAL library
+- `P2`: Run if upgrading major version of Azure.Identity and/or MSAL library
+
+## Azure Global Instance
+
+Azure Global Instance is the most important Azure instance, all tests should run against it.
+
+### PWSH Platform Matrix
+
+In theory all the combination of different OS platforms and PWSH versions should be covered, to compromise time effort and platform coverages, we should at least cover the following platforms. (For different versions of pwsh 7, need to cover at least smallest and biggest patch version for each major.minor version, currently it should be 7.0.0, 7.0.3 and 7.1.0. The reason to cover 7.0.0 is the future version of Azure.Core may contain higher version of built-in assemblies of pwsh.)
+
+- Windows PowerShell 5.1
+- PWSH 7.0.0 on Windows
```

Comment:
Do we need to cover PwoerShell 7.0 and 7.1 both? I worry about the long list will distract our focus.

Reply:
We need to consider PowerShell version on SAW machine. Locally we can test it via setting ConstrainedLanguage

Reply:
PS 7.1 should be covered because it is latest version and based on .NET 5.
PS 7.0 should be covered because it is LTS version and based on .net core 3.
Based on the telemetry data the usage of PS 7.0 is 500 times of PS 7.1, I suggest to give higher priority to PS 7.0.

As to SAW machine, it is covered in later section with P2 priority.

Reply:
The next paragraph has stated which tests should be run on what platforms.